lib/client/ipam_block_reader_writer.go

```diff
@@ -139,6 +124,20 @@ func (rw blockReaderWriter) claimNewAffineBlock(
 	return nil, noFreeBlocksError("No Free Blocks")
 }
 
+// checkIPPoolCIDR checks if the IP Pool that is passed in belongs to the list of IP Pools
+// that should be used for assigning IPs from.
+func checkIPPoolCIDR(pool api.IPPool, givenPools []cnet.IPNet) bool {
```

Comment:
I was actually thinking that `givenPools` seems like an odd name (I realize this was pre-existing).  I wonder if we should rename to `requestedPools`  and then this method could be something like

isIPPoolInRequestedPools().

Or rather than passing in an IPPool() you could pass in the CIDR and then have a more generic method that checks if a CIDR is in a slice of CIDRs.  I think this is a nicer interface since it doesn't mix []IPNet and IPPool.